### ISSUE TITLE
Inquiry About Auto-Tuning Feature in GETI

### ISSUE THREAD

**Issue by yisu-intel** [open-edge-platform/geti_v2#461]
### 📄 Description

Hello, and thank you for the great work your team has done on GETI.

Recently, my manager expressed interest in adding some new features to GETI, specifically around auto-tuning, i.e., hyperparameter optimization. We’ve been looking into open-source solutions such as Hyperopt and Ray Tune, and I wanted to reach out to ask:

Would your team be interested in incorporating such an auto-tuning feature into GETI?

### 🎯 Objective

auto-tuning

### 🧑‍💻 User Story

hyperparameter optimization

### 📌 Requirements Covered

Given a set of important hyperparameters, such as learning rate and batch size, the GETI platform can automatically search for the optimal hyperparameters.

**Comment by BrittvanAkker** [comment]
Hi Yisu, thanks for reaching out. We supported Hyper Parameter Optimization (HPO) in earlier versions of Geti. However, HPO is hard to generalize across all algorithms inside Geti, which led in some situations to bad performance results. This was confusing for our users. The scalability of HPO was a bottleneck as well: when adding new algorithms to our model suite, a lot of testing was required to ensure the quality of this optimization option. We could not guarantee the testing coverage that was required to ensure the quality of this capability.

To avoid misleading users with inconsistent performance results and to keep the system scalable, we decided to remove HPO from Geti. Users can still take our [training backend](https://github.com/open-edge-platform/training_extensions) and utilize any hyper parameter optimization library that meets their needs, but we will not add HPO back to the Geti platform as there has not been any customer request for this capability. 

I hope I answered your question, please let me know if anything remains unclear.

**Comment by yisu-intel** [comment]
Thank you for the detailed explanation. I understand the reasoning behind the decision now. Appreciate your clarification!
